### ISSUE TITLE
Discover new __consumer_offsets partitions

### ISSUE THREAD

**Issue by thomaslee** [linkedin/Burrow#326]
Perhaps an unlikely use case, but I made the mistake of starting Burrow before running any consumers during some local testing in a small docker-compose environment & lost a lot of time trying to figure out why Burrow wasn't discovering new consumer groups. I _think_ it boils down to this log message:

```
{"level":"info","ts":1516057630.5676234,"msg":"starting consumers","type":"module","coordinator":"consumer","class":"kafka","name":"eventbus","topic":"__consumer_offsets","count":0}
```

Notice the `count` field, which is zero because the __consumer_offsets topic does not exist at the time Burrow starts. After running a consumer to force creation of `__consumer_offsets` & restarting Burrow, it starts detecting new consumer groups as expected. Again, the hint is in the log message:

```
{"level":"info","ts":1516057932.7690308,"msg":"starting consumers","type":"module","coordinator":"consumer","class":"kafka","name":"eventbus","topic":"__consumer_offsets","count":50}
```

I sort of half suspect Burrow might also silently ignore new `__consumer_offsets` partitions even if the topic exists, but but haven't had a chance to check.

Would be nice if we could either flame out hard if no `__consumer_offsets` topic is present or (perhaps ideally) automatically discover any changes to `__consumer_offsets`.

**Comment by toddpalino** [comment]
Yeah, this is a definite issue in the way the consumer starts up. It does it by individually starting a PartitionConsumer for each partition, and it doesn't have a function to watch for changes to the topic to pick up new partitions. It would be sufficient to add code into the kafka client module to periodically check for a change in the partition count for the topic and spin up new PartitionConsumers in that case.

As a somewhat orthogonal note, it's generally a **really** bad idea to change the number of partitions for `__consumer_offsets` after it is initially created. This causes the hashing for consumer group name to partition to change, which means the  group coordinator will likely have no idea of the history (i.e. offset commits) for the group.
